.github/workflows/tree.yml

```diff
@@ -46,6 +46,9 @@ jobs:
       - name: Check entrypoint tag
         run: ./tools/check-entrypoint-tag.sh
 
+      - name: Check downgrade versions
+        run: ./tools/check-downgrade-versions.sh
+
```

Comment:
AFAIU this workflow won't run on pushing a tag.

Reply:
Fixed. We have do some lifting though to keep tag from changing on checkout action.

```diff
diff --git a/.github/workflows/tree.yml b/.github/workflows/tree.yml
index 74d15a77a..0d841a635 100644
--- a/.github/workflows/tree.yml
+++ b/.github/workflows/tree.yml
@@ -5,6 +5,8 @@ on:
     branches:
       - 'master'
       - 'release/**'
+    tags:
+      - '**'

 concurrency:
   # Update of a developer branch cancels the previously scheduled workflow
@@ -42,6 +44,12 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          # If workflow is triggered by pushing a tag then and we checkout
+          # using github.ref then checkout action overwrite tag with tag
+          # without annotation. This breaks check-downgrade-versions.sh
+          # script as it is triggered only on annotated tags. Let's
+          # checkout by sha in this case.
+          ref: ${{ github.reftype == 'tag' && github.sha || github.ref }}

       - name: Check entrypoint tag
         run: ./tools/check-entrypoint-tag.sh
```